File: examples/java/src/EnNluExample2.java
import tencent.ai.texsmart.*;

public class EnNluExample2 {
    public static void main(String[] args) {
    	String userDir = System.getProperty("user.dir");
		String dataDir = args.length >= 1 ? args[0] : userDir + "/../../data/nlu/kb/";
		//if(args.length < 1) {
		//	System.out.println("Please specify the data folder for the NLU engine");
		//	return;
		//}
	
		System.out.println("Creating and initializing the NLU engine...");
		NluEngine engine = new NluEngine();
		int workerCount = 4;
		boolean ret = engine.init(dataDir, workerCount);
		if(!ret) {
			System.out.println("Failed to initialize the engine");
			return;
		}

		String options = "{\"ner\":{\"enable\":true,\"fine_grained\":false}}";
		System.out.printf("Options: %s\n", options);

    	System.out.println("=== Parse a piece of English text ===");
    	//String text = "The history of natural language processing generally started in the 1950s.";
    	String text = "John Smith stayed in San Francisco last month.";
    	NluOutput output = engine.parseText(text, options);

    	System.out.printf("Input text: %s\n", text);
    	System.out.printf("Output norm text: %s\n", output.normText());
    	
    	System.out.println("Word-level segmentation results:");
		for(NluOutput.Term word : output.words()) {
			System.out.printf("\t%s\t(%d,%d)\t%s\n", word.str, word.offset, word.len, word.tag);
		}

		System.out.println("Phrase-level segmentation results:");
		for(NluOutput.Term phrase : output.phrases()) {
			System.out.printf("\t%s\t(%d,%d)\t%s\n", phrase.str, phrase.offset, phrase.len, phrase.tag);
		}
	
		System.out.println("NER results:");
		for(NluOutput.Entity ent : output.entities()) {
			System.out.printf("\t%s\t(%d,%d)\t%s\t%s\n", ent.str, ent.offset, ent.len, ent.type.name, ent.meaning);
		}
    }
}

File: examples/python/en_nlu_example2.py

#!/usr/bin/env python
#encoding=utf-8
import sys
import os.path
module_dir = os.path.dirname(os.path.realpath(__file__))
sys.path.append(module_dir+'/../../lib/')
from tencent_ai_texsmart import *

print('##################################################')
print('# Example-2: Parsing text using options')
print('##################################################')

print('Creating and initializing the NLU engine...')
engine = NluEngine(module_dir + '/../../data/nlu/kb/', 1)

#disable fine-grained NER:
print('Options: Enable NER but disable fine-grained NER')
options = '{"ner\":{"enable":true,"fine_grained":false}}'

print('=== Parse a piece of English text ===')
output = engine.parse_text_ext("John Smith stayed in San Francisco last month.", options)
print(u'Norm text: %s' % output.norm_text())
print('Word-level segmentation results:')
for word in output.words():
    print(u'\t%s\t(%d,%d)\t%s' % (word.str, word.offset, word.len, word.tag))
print('Phrase-level segmentation results:')
for phrase in output.phrases():
    print(u'\t%s\t(%d,%d)\t%s' % (phrase.str, phrase.offset, phrase.len, phrase.tag))
print('NER results:')
for entity in output.entities():
    print(u'\t%s\t(%d,%d)\t%s\t%s' % (entity.str, entity.offset, entity.len, entity.type.name, entity.meaning))
